src/kvstore.c

```diff
@@ -583,13 +643,43 @@ void kvstoreIteratorRelease(kvstoreIterator *kvs_it) {
     hashtableIterator *iter = &kvs_it->di;
     hashtableResetIterator(iter);
     /* In the safe iterator context, we may delete entries. */
-    freeHashtableIfNeeded(kvs_it->kvs, kvs_it->didx);
+    if (kvs_it->didx != -1) {
+        freeHashtableIfNeeded(kvs_it->kvs, kvs_it->didx);
+    }
+    if (kvs_it->importing_iter) {
+        hashtableReleaseIterator(kvs_it->importing_iter);
+    }
     zfree(kvs_it);
 }
 
+static int kvstoreIteratorNextImportingHashtableIndex(kvstoreIterator *kvs_it) {
+    if (kvs_it->importing_iter == NULL) {
+        kvs_it->importing_iter = hashtableCreateIterator(kvs_it->kvs->importing, 0);
+    }
+    intptr_t didx;
+    while (hashtableNext(kvs_it->importing_iter, (void **)&didx)) {
+        if (kvstoreHashtableSize(kvs_it->kvs, didx)) {
+            return didx;
+        }
+    }
+    return -1;
```

Comment:
nit. use the new symbolic name `KVSTORE_INDEX_NOT_FOUND`?
```suggestion
    return KVSTORE_INDEX_NOT_FOUND;
```